lib/util.js

```diff
@@ -18,12 +20,41 @@ function hmac (key, message) {
   return h.digest()
 }
 
+function toU32BE (n) {
+  const bn = new BigNumber(n)
+  if (bn.lt('0') || bn.gte(MAX_U32)) {
+    throw new Error('number out of range for u32. n=' + n)
+  }
+
+  const buf = Buffer.alloc(4)
+  buf.writeUInt32BE(bn.toNumber(), 0)
+  return buf
+}
+
+function toU64BE (n) {
+  const bn = new BigNumber(n)
+  if (bn.lt('0') || bn.gte(MAX_U64)) {
```

Comment:
Better to check that `n` is in the valid range before turning it into a big number